data-prepper-plugins/dynamodb-source/src/test/java/org/opensearch/dataprepper/plugins/source/dynamodb/converter/StreamRecordConverterTest.java

```diff
@@ -188,7 +188,8 @@ void test_writeSingleRecordToBuffer() throws Exception {
             "and/or",
             "c:\\Home",
             "I take\nup multiple\nlines",
-            "String with some \"backquotes\"."
+            "String with some \"backquotes\".",
+            "String with some control characters: \0\1\2\3\4\5\6\7\10\11\12\13\14\15\16\17\20\21\22\23\24\25\26\27\28\29\30\31\127\b\t\n\f\r"
```

Comment:
We should also have a test that provides both the input and the expected output to verify that the result is what we want.